src/main/java/reactor/core/publisher/FluxWindow.java

```diff
@@ -748,64 +719,40 @@ public void dispose() {
 		}
 
 		@Override
-		public Object downstream() {
+		public Subscriber<? super Flux<T>> actual() {
 			return actual;
 		}
 
 		@Override
-		public boolean isCancelled() {
-			return cancelled;
-		}
-
-		@Override
-		public boolean isStarted() {
-			return s != null && !done && !cancelled;
-		}
-
-		@Override
-		public boolean isTerminated() {
-			return done;
-		}
-
-		@Override
-		public Object upstream() {
-			return s;
-		}
-
-		@Override
-		public Throwable getError() {
-			return error;
-		}
-
-		@Override
-		public long expectedFromUpstream() {
-			return (size + skip) - produced;
-		}
-
-		@Override
-		public long limit() {
-			return skip;
+		public boolean isDisposed() {
+			return cancelled || done;
 		}
 
 		@Override
-		public Iterator<?> downstreams() {
-			return Arrays.asList(toArray())
-			             .iterator();
-		}
-
-		@Override
-		public long downstreamCount() {
-			return size();
-		}
-
-		@Override
-		public long getCapacity() {
-			return size;
+		public Object scan(Attr key) {
+			switch (key) {
+				case PARENT:
+					return s;
+				case CANCELLED:
+					return once == 1;
+				case CAPACITY:
+					return size;
+				case TERMINATED:
+					return done;
+				case BUFFERED:
+					return queue.size() + size();
```

Comment:
I'm not 100% sure, as a window can be in both queues, but I fallback to the direct buffered reference, individual windows can ell how much they buffered then.